components/formats-gpl/src/loci/formats/in/OIRReader.java

```diff
@@ -0,0 +1,1238 @@
+import loci.common.xml.XMLTools;
+import loci.formats.CoreMetadata;
+import loci.formats.FormatException;
+import loci.formats.FormatReader;
+import loci.formats.FormatTools;
+import loci.formats.MetadataTools;
+import loci.formats.meta.MetadataStore;
+import ome.xml.model.enums.Immersion;
+import ome.xml.model.primitives.Color;
+import ome.xml.model.primitives.Timestamp;
+import ome.units.UNITS;
+import ome.units.quantity.Length;
+
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * OIRReader is the file format reader for Olympus .oir files.
+ */
+public class OIRReader extends FormatReader {
+
+  // -- Constants --
+
+
+  // -- Fields --
+
+  private HashMap<String, PixelBlock> pixelBlocks = new HashMap<String, PixelBlock>();
+  private ArrayList<Channel> channels = new ArrayList<Channel>();
+  private ArrayList<Laser> lasers = new ArrayList<Laser>();
+  private ArrayList<Detector> detectors = new ArrayList<Detector>();
+  private ArrayList<Objective> objectives = new ArrayList<Objective>();
+  private Length physicalSizeX;
+  private Length physicalSizeY;
+  private Length physicalSizeZ;
+  private Timestamp acquisitionDate;
+  private int defaultXMLSkip = 36;
+  private int blocksPerPlane = 0;
+  private String[] pixelUIDs = null;
+  private String baseName;
+  private int lastChannel = -1;
+
+  // -- Constructor --
+
+  /** Constructs a new OIR reader. */
+  public OIRReader() {
+    super("Olympus OIR", "oir");
+    domains = new String[] {FormatTools.LM_DOMAIN};
+  }
+
+  // -- IFormatReader API methods --
+
+  /* @see loci.formats.IFormatReader#get8BitLookupTable() */
+  @Override
+  public byte[][] get8BitLookupTable() throws FormatException, IOException {
+    int pixelType = getPixelType();
+    if ((pixelType != FormatTools.INT8 && pixelType != FormatTools.UINT8) ||
+      !isIndexed())
+    {
+      return null;
+    }
+    if (lastChannel < 0 || lastChannel >= channels.size()) {
+      return null;
+    }
+    return (byte[][]) channels.get(lastChannel).lut;
+  }
+
+  /* @see loci.formats.IFormatReader#get16BitLookupTable() */
+  @Override
+  public short[][] get16BitLookupTable() throws FormatException, IOException {
+    int pixelType = getPixelType();
+    if ((pixelType != FormatTools.INT16 && pixelType != FormatTools.UINT16) ||
+      !isIndexed())
+    {
+      return null;
+    }
+    if (lastChannel < 0 || lastChannel >= channels.size()) {
+      return null;
+    }
+    return (short[][]) channels.get(lastChannel).lut;
+  }
+
+
+  /**
+   * @see loci.formats.IFormatReader#openBytes(int, byte[], int, int, int, int)
+   */
+  @Override
+  public byte[] openBytes(int no, byte[] buf, int x, int y, int w, int h)
+    throws FormatException, IOException
+  {
+    FormatTools.checkPlaneParameters(this, no, buf.length, x, y, w, h);
+
+    int nextPointer = 0;
+    int blocksPerChannel = pixelBlocks.size() / channels.size();
+
```

Comment:
Leaving both of those lines in was an oversight, so will certainly remove in the forthcoming follow-up PR.  It worries me a little that Findbugs didn't catch either of them, though, since it seems like both should have been flagged as dead local stores.